mantidproject/mantid
# Indirect Bayes - BayesStretch algorithm

The functionality from the Quest function in the IndirectBayes.py script is replaced by this mantid algorithm. The interface has been updated to use this algorithm and there is an accompanying test and documentation.

The `RunQuest` function in the IndirectBayes.py script should be removed once the we have confirmed that no one is using it.
# To test

**Tests must be performed on windows due to reliance on compiled F2Py modules.**

The following script should test the functionality of the algorithm:
_`irs26176_graphite002_red` and `irs26173_graphite002_res` can be obtained from the unit test directory_

```
irs26176_graphite002_red = Load('irs26176_graphite002_red.nxs')
irs26173_graphite002_res = Load('irs26173_graphite002_res.nxs')

 # Run BayesStretch algorithm
fit_group, contour_group = BayesStretch(SampleWorkspace=irs26176_graphite002_red, ResolutionWorkspace=irs26173_graphite002_res, 
                                                        EMin=-0.2, EMax=0.2, Background='Sloping', Loop=True)

```
- Ensure this script produces 2 workspace groups 
  - The group suffixed with `_Contour` should contain 10 workspaces suffixed with `_Zp<WorkspaceIndex>`
  - The group suffixed with `_Fit` should contain the 2 workspaces (one suffixed with `_Sigma` the other `_Beta`)

Using the same sample and resolution workspaces as those defined in the script, try using the interface to perform the same operation:
- Open Stretch (Interfaces > Indirect > Bayes > Stretch)
- Add the sample (irs26176) and resolution (irs26173)
- Check the `Sequential Fit` option
- Change the `Plot Result` (combo box in the bottom left) to `All`
- Check Save Result
- Click `Run`
- Ensure the algorithm runs without error and two plots are produced (see below)
- Ensure that the workspace groups produced by the algorithm have been saved to your default save directory.

![image](https://cloud.githubusercontent.com/assets/13132128/16983936/6a8b3bec-4e6f-11e6-872a-65b9b698d863.png)

![image](https://cloud.githubusercontent.com/assets/13132128/16983939/6e39b066-4e6f-11e6-9ade-82871428a990.png)

Fixes #13956 

[RELEASE NOTES](https://github.com/mantidproject/mantid/blob/a37025b5a040fdfb5ea27d0e58e3fcac3d124c44/docs/source/release/v3.8.0/indirect_inelastic.rst)

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
##### Functional Tests
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
